app/src/main/java/com/demoapp/viewmodel/MainActivityViewModel.kt

```diff
@@ -1,27 +1,23 @@
 package com.demoapp.viewmodel
 
-import android.app.Application
-import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import com.demoapp.App
-import com.demoapp.retrofit.RetrofitInterface
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.demoapp.database.DataDuo
 import com.demoapp.database.DataRepository
 import com.demoapp.model.DataEntity
+import com.demoapp.retrofit.RetrofitInterface
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-class MainActivityViewModel(application: Application) : AndroidViewModel(application) {
-
-    @Inject
-    lateinit var dataDuo: DataDuo
+class MainActivityViewModel @Inject constructor(dataDuo: DataDuo, mService: RetrofitInterface) :
```

Comment:
We don't need direct access to Dao and Retrofit in viewModel. it should be in [Repository](https://developer.android.com/jetpack/guide) class.